validator/app/src/compute_horde_validator/validator/facilitator_client.py

```diff
@@ -194,29 +197,39 @@ async def wait_for_specs(self):
         specs_queue = []
         while True:
             validator_hotkey = settings.BITTENSOR_WALLET().hotkey.ss58_address
-            msg = await self.channel_layer.receive(self.channel_name)
-            specs = MachineSpecsUpdate(
-                specs=msg["specs"],
-                miner_hotkey=msg["miner_hotkey"],
-                batch_id=msg["batch_id"],
-                validator_hotkey=validator_hotkey,
-            )
-            logger.debug(f"sending machine specs update to facilitator: {specs}")
-
-            if self.ws is None:
+            try:
+                msg = await asyncio.wait_for(
+                    self.channel_layer.receive(self.channel_name), timeout=20 * 60
+                )
+
+                specs = MachineSpecsUpdate(
+                    specs=msg["specs"],
+                    miner_hotkey=msg["miner_hotkey"],
+                    batch_id=msg["batch_id"],
+                    validator_hotkey=validator_hotkey,
+                )
+                logger.debug(f"sending machine specs update to facilitator: {specs}")
+
                 specs_queue.append(specs)
-            else:
-                while len(specs_queue) > 0:
-                    await self.send_model(specs_queue.pop(0))
-                await self.send_model(specs)
+                if self.ws is not None:
+                    while len(specs_queue) > 0:
+                        spec_to_send = specs_queue.pop(0)
+                        try:
+                            await self.send_model(spec_to_send)
+                        except Exception as exc:
+                            logger.error("Error occurred while sending specs", exc_info=exc)
+                            specs_queue.insert(0, spec_to_send)
+                            break
+            except TimeoutError:
+                logger.debug("wait_for_specs still running")
```

Comment:
I think this log is a bit confusing.

```suggestion
                logger.debug("Timed out while waiting for machine specs after {X} seconds. Trying again...")
```

Reply:
added the timeout just to verify that the tasks still runs to see if the bug re-appears - will remove it after I confirm is fixed